src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```
@@ -0,0 +1,625 @@
+     * @throws FileManagerException      if any other problem happened
+     */
+    @Override
+    public void createFile(String name) throws FileAlreadyExistException, FileManagerException {
+        File file = new File(name);
+
+        try {
+            if (file.isDirectory()) {
+                throw new IOException();
+            }
+            boolean fcnf = file.createNewFile();
+            if (fcnf) {
+                System.out.println("File has been created successfully");
+            } else {
+                throw new FileAlreadyExistException(file);
+            }
+        } catch (FileAlreadyExistException n) {
+            System.out.println(n.getMessage());
+        } catch (FileManagerException n) {
+            System.out.println(n.getMessage());
+        } catch (IOException e) {
+            System.out.println("Not File");
+        }
+    }
+
+
+    @Override
+    public void createDirectory(String name) throws DirectoryAlreadyExistException, FileManagerException {
+        File file = new File(name);
+        try {
+            if (file.isDirectory()) {
+                throw new IOException();
+            }
+            boolean fcnf = file.mkdir();
+            if (fcnf) {
+                System.out.println("Directory has been created successfully");
+            } else {
+                throw new DirectoryAlreadyExistException( file);
+            }
+        } catch (DirectoryAlreadyExistException n) {
+            System.out.println(n.getMessage());
+        } catch (FileManagerException n) {
+            System.out.println(n.getMessage());
+        } catch (IOException e) {
+            System.out.println("Not Directory");
+        }
+
+    }
+
+    @Override
+    public void changeDirectory(String name) throws IllegalDirectoryException {
+
+        try {
+            Paths.get(name);
+            setPath(name);
+            System.out.println("Changed Directory");
+        } catch (IllegalDirectoryException ide) {
```

You should throw this exception if smth goes wrong (for example someone tries to give you a File instead of Directory)